### PR TITLE
Session management

### PR DESCRIPTION
**Pull request by MichaelCalabro**

### CODE REVIEW

File: server/SessionManager.js [hunk 1]
```diff
@@ -0,0 +1,47 @@
+
+const roomSocketIds = new Map();
+const sessions = new Map();
+
+
+export function registerSession(host, apiSearchParameters, socketId){
+
+    var session = new Session(host, apiSearchParameters);
+    var roomCode = generateRoomCode(6);
```
Comment (ahmedsakr) [inline server/SessionManager.js:9]: make a constant for the `6` here

File: server/SessionManager.js [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+    roomSocketIds.set(roomCode, socketId);
+    sessions.set(socketId, session);
+
+    return roomCode;
+}
+
+export function joinSession(roomCode, user){
+
+    if(roomSocketIds.has(roomCode)){
+        sessions.get(roomSocketIds.get(roomCode)).addUser(user);
```
Comment (ahmedsakr) [inline server/SessionManager.js:20]: good idea to unroll this line into a few lines, too much getting happening here

File: server/SessionManager.js [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+    if(roomSocketIds.has(roomCode)){
+        
+        //Start session, get restaurants and add them to session object
+
+        return true;
+    }
+    return false;
+}
+
+export function generateRoomCode(length) {
```
Comment (ahmedsakr) [inline server/SessionManager.js:37]: like we discussed, check if there is an existing library to do this job

File: server/Session.js [hunk 4]
```diff
@@ -0,0 +1,43 @@
...
+        this.rightSwipes = new Map();
+        this.apiSearchParameters = searchParameters;
+
+    }
+
+    addUser(user){
+        this.users.push(user);
+    }
+
+    getUsers(){
```
Comment (ahmedsakr) [inline server/Session.js:15]: we don't really need object `getters` in javascript, they aren't hidden like Java.

File: server/Session.js [hunk 5]
```diff
@@ -0,0 +1,43 @@
...
+
+    constructor(host, searchParameters) {
+        this.users = [host];
+        this.restaurants = [];
+        this.rightSwipes = new Map();
+        this.apiSearchParameters = searchParameters;
+
+    }
+
+    addUser(user){
```
Comment (ahmedsakr) [inline server/Session.js:11]: should check if the user is already in this lobby

**Comment by ahmedsakr** [comment]
you're going to need to update from main

File: server/Session.js [hunk 6]
```diff
@@ -0,0 +1,36 @@
...
+    swipeAndCheckMatch(restaurantId) {
+
+        if (!this.rightSwipes.has(restaurantId)) {
+            this.rightSwipes.set(restaurantId, 1);
+        }
+        else {
+            this.rightSwipes.set(restaurantId, this.rightSwipes.get(restaurantId) + 1);
+        }
+
+        if(this.rightSwipes.get(restaurantId) >= this.users.length){
```
Comment (ahmedsakr) [inline server/Session.js:30]: line 30-34 can be simplified to

`return this.rightSwipes.get(restaurantId) >= this.users.length;`

Comment (ahmedsakr) [inline server/Session.js:30]: line 30-34 can be simplified to `return this.rightSwipes.get(restaurantId) >= this.users.length;`

File: server/SessionManager.js [hunk 7]
```diff
@@ -0,0 +1,59 @@
+import Session from "./Session";
+var roomCodeGenerator = require("randomstring")
+
+const roomSocketIds = new Map();
+const sessions = new Map();
+const roomCodeLength = 6;
```
Comment (ahmedsakr) [inline server/SessionManager.js:6]: constants should be UPPER_CAMEL_CASE: `ROOM_CODE_LENGTH` and `ROOM_CODE_CHARSET`

File: server/Session.js (hunk 6 above)
Comment (MichaelCalabro) [reply server/Session.js:30]: good catch

**Review by ahmedsakr — APPROVED** [review]

**Review by joshcampitelli — APPROVED** [review]
Looks good 👍

**Pull request merged**